trace-obfuscation/src/replacer.rs

```diff
@@ -0,0 +1,198 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0. This product includes software
+// developed at Datadog (https://www.datadoghq.com/). Copyright 2023-Present
+// Datadog, Inc.
+
+use crate::pb;
+use regex::Regex;
+
+pub trait TraceTagReplacer {
+    fn replace_trace_tags(trace: &mut [pb::Span], rules: &[ReplaceRule]);
+}
+
+#[derive(Debug)]
+pub struct ReplaceRule<'a> {
+    // name specifies the name of the tag that the replace rule addresses. However,
+    // some exceptions apply such as:
+    // • "resource.name" will target the resource
+    // • "*" will target all tags and the resource
+    name: &'a str,
+
+    // re holds the regex pattern for matching.
+    re: regex::Regex,
+
+    // repl specifies the replacement string to be used when Pattern matches.
+    repl: &'a str,
+}
+
+struct DefaultTraceTagReplacer {}
```

Comment:
I debated having a `rules: Vec<ReplaceRules>` field in `DefaultTraceTagReplacer` (along with a constructor), but I reasoned that the `ReplaceRule`s will be kept as a field in the Agent itself, given that the Agent has to process the user-set configuration.